mithril-stm/src/schnorr_signature/mod.rs

```diff
@@ -0,0 +1,173 @@
+// TODO: Remove
+#![allow(dead_code)]
+
+use midnight_circuits::{
+    ecc::{hash_to_curve::HashToCurveGadget, native::EccChip},
+    hash::poseidon::PoseidonChip,
+    types::AssignedNative,
+};
+use midnight_curves::{
+    Fq as JubjubBase, Fr as JubjubScalar, JubjubAffine, JubjubExtended, JubjubSubgroup,
+};
+use sha2::{Digest, Sha256};
+
+use anyhow::{Result, anyhow};
+
+mod signature;
+mod signing_key;
+mod verification_key;
+
+use signature::*;
+// use signing_key::*;
+use verification_key::*;
+
+/// A DST to distinguish between use of Poseidon hash
+pub const DST_SIGNATURE: JubjubBase = JubjubBase::from_raw([0u64, 0, 0, 0]);
+pub const DST_LOTTERY: JubjubBase = JubjubBase::from_raw([1u64, 0, 0, 0]);
+
+/// Defining a type for the CPU hash to curve gadget
+type JubjubHashToCurve = HashToCurveGadget<
+    JubjubBase,
+    JubjubExtended,
+    AssignedNative<JubjubBase>,
+    PoseidonChip<JubjubBase>,
+    EccChip<JubjubExtended>,
+>;
+
+/// Convert an arbitrary array of bytes into a Jubjub scalar field element
+/// First hash the message to 256 bits use Sha256 then perform the conversion
+/// TODO: Handle the unwrap properly
+pub(crate) fn hash_msg_to_jubjubbase(msg: &[u8]) -> Result<JubjubBase> {
+    let mut hash = Sha256::new();
+    hash.update(msg);
+    let hmsg = hash.finalize();
+    let mut output = [0u8; 32];
+    // Adding a check here but this
+    if hmsg.len() == output.len() {
+        output.copy_from_slice(&hmsg);
+    } else {
+        return Err(anyhow!(
+            "Hash of the message does not have the correct lenght."
+        ));
+    }
+    Ok(JubjubBase::from_raw([
+        u64::from_le_bytes(output[0..8].try_into()?),
+        u64::from_le_bytes(output[8..16].try_into()?),
+        u64::from_le_bytes(output[16..24].try_into()?),
+        u64::from_le_bytes(output[24..32].try_into()?),
+    ]))
+}
+
+/// Extract the coordinates of a given point
+/// This is mainly use to feed the Poseidon hash function
+pub(crate) fn get_coordinates(point: JubjubSubgroup) -> (JubjubBase, JubjubBase) {
+    let extended = JubjubExtended::from(point); // Convert to JubjubExtended
+    let affine = JubjubAffine::from(extended); // Convert to JubjubAffine (affine coordinates)
+    let x = affine.get_u(); // Get x-coordinate
+    let y = affine.get_v(); // Get y-coordinate
+    (x, y)
```

Comment:
All of these comments are not needed and could be removed with variable names which are more explicit:
```suggestion
    let extended = JubjubExtended::from(point); 
    let affine = JubjubAffine::from(extended);
    let x = affine.get_u(); 
    let y = affine.get_v(); 
    (x, y)
```